woocommerce/woocommerce-android
# Use WCOrderModelTable when offline to load cached orders

~Fixes #1656 by bypassing the use of the `remoteOrderIds` provided by the list `ListStore` to the `in the `OrderListItemDatasource` in the following methods:~

~**`getItemIdentifiers(listDescriptor, remoteItemIds, isListFullyFetched)`**~
~Instead of using the provided `remoteItemIds` argument, I've updated the code to instead use the `listDescriptor` to query the cached records in `WCOrderModel`. This ensures that if a list has never been fetched, but matching orders do exist in cache, we return those cached orders.~

~**`getItemsAndFetchIfNecessary(listDescriptor, itemIdentifiers)`**~
~Added a check to only attempt to fetch orders if the network is connected. This wasn't necessary, but it's much cleaner and prevents the "error fetch order" snack message from being displayed repeatedly.~

**Note** - The following PRs must be merged before this PR can be merged:
- ~https://github.com/woocommerce/woocommerce-android/pull/1559~
- ~FluxC: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1444~

Had a meeting with @malinajirka to washout possible issues. Turns out I won’t be able to use the `listDescriptor` to query the db when offline due to the fact that we don’t delete any records from `WCOrderModel` at this time. So that table may contain orders that no longer exist. I've closed that FluxC PR and reverted my offline changes in this PR. 

I’ve added new logic to fetch both the `ALL` and `PROCESSING` lists at the same time.  So if the device is online, both lists will have data. I’m also prefetching order statuses and payment gateways for the same reason.

## To Test
1. Do a fresh install of the app and login
2. Click on the “Orders” tab - the “ALL” tab will populate with orders
3. Enable airplane mode on the device
4. Click on the “PROCESSING” tab - processing orders should populate.
5. Click on the search icon - order status list should also populate. 
6. Click on an order status other than “Processing” - the network error empty state should be displayed. This is because this list was never fetched and is expected.
